### DISCUSSION TITLE
finalized block doesn't update when rolling up to an L2

### DISCUSSION THREAD

**Discussion by arshsingh** [ethereum-optimism/developers#580] [Deploying the OP Stack]
We're deploying an op-stack chain as an L3 rolling up to Base, but the l2_finalized block only updates when op-node is restarted.

The value of [finalityLookback](https://github.com/ethereum-optimism/optimism/blob/fc662d4a3ce28b76566c4703e32d731be69a8524/op-node/rollup/derive/engine_queue.go#L133) is currently hard-coded to `4*32 + 1`, which If my understanding is correct doesn't work well when rolling up to an L2, because a finalization event there can take `4*32*6` blocks, or longer depending on the L2's configuration.

Deploying with `v1.7.4`, I keep seeing the following error in the logs:
```
received L1 finality signal, but missing data for immediate L2 finalization
```

I deployed op-node with an increased value for finalityLookback and I don't see this issue anymore. Blocks are finalizing and the error message disappeared from the logs.

Would it be possible to make `finalityLookback` configurable?

**Comment by dindonero** [comment]
I'm trying to do the same thing but every time I try to bridge ETH into the L3, my op-node logs the `received L1 finality signal, but missing data for immediate L2 finalization` message. After a few seconds I get `processing L1 finality information` so I imagine it's processing the l1 (l2 in this case) information as finalized. But it never updates my account's balance on the L3.

Also, op-geth updates chain head with a new block about 5 min later, so it seems it should be working.
